app/controllers/concerns/mhv_controller_concerns.rb

```diff
@@ -12,13 +12,16 @@ module MHVControllerConcerns
   protected
 
   def authorize
+    # The user must (1) have an account that is either existing or
+    # created and upgraded by us or (2) be eligible to have an
+    # account created and upgraded by us.
     raise_access_denied unless accessible_or_eligible_for_creation?
+
+    # Stop if the user needs to accept terms and conditions.
     raise_requires_terms_acceptance if current_user.mhv_account.needs_terms_acceptance?
-    begin
-      current_user.mhv_account.create_and_upgrade! unless current_user.mhv_account.accessible?
-    ensure
-      raise_something_went_wrong unless current_user.mhv_account.accessible?
-    end
+
+    # Stop if further actions are necessary to access MHV services.
+    raise_something_went_wrong unless authorized?
```

Comment:
Changed to `authorized?`, a protected method. Included this concerns in `MhvAccountsController`, which can modify `authorized?` to instead check for `eligible?` rather than `accessible?` for the `#create` action.